src/download.ts

```diff
@@ -234,22 +234,28 @@ export async function getMetadata(
  * @returns - The content of the file.
  * @throws - Will throw if the skylinkUrl does not contain a skylink or if the path option is not a string.
  */
-export async function getFileContent(
+export async function getFileContent<T = unknown>(
   this: SkynetClient,
   skylink: string,
   customOptions?: CustomDownloadOptions
-): Promise<Record<string, unknown>> {
+): Promise<T> {
   const opts = { ...defaultDownloadOptions, ...this.customOptions, ...customOptions };
   const url = this.getSkylinkUrl(skylink, opts);
 
   // GET request the skylink
-  const response = await this.executeRequest({
+  const { data } = await this.executeRequest({
     ...opts,
     method: "get",
     url,
   });
 
-  return response.data;
+  if (typeof data === "undefined") {
+    throw new Error(
+      "Did not get 'data' in response despite a successful request. Please try again and report this issue to the devs if it persists."
+    );
+  }
+
+  return data;
```

Comment:
Would it make sense to take this one step further and verify that `data` is `string`? Or we want to keep this intentionally open and let the caller make the assertion, so we can potentially add other data types as well?

Reply:
Maybe I should have read the existing discussion before reading the code. https://github.com/NebulousLabs/skynet-js/pull/227#issuecomment-742692870 resolves this.